lib/internal/repl/utils.js

```diff
@@ -87,7 +113,137 @@ function isRecoverableError(e, code) {
   }
 }
 
+function setupPreview(repl, contextSymbol, bufferSymbol, active) {
+  // Simple terminals can't handle previews.
+  if (process.env.TERM === 'dumb' || !active) {
+    return { showInputPreview() {}, clearPreview() {} };
+  }
+
+  let preview = '';
+  let lastPreview = '';
+
+  const clearPreview = () => {
+    if (preview) {
+      moveCursor(repl.output, 0, 1);
+      clearLine(repl.output);
+      moveCursor(repl.output, 0, -1);
+      preview = '';
+    }
+  };
+
+  // This returns a code preview for arbitrary input code.
+  function getPreviewInput(input, callback) {
+    sendInspectorCommand((session) => {
+      session.post('Runtime.evaluate', {
```

Comment:
One more thing I noticed while playing around is that, `util.inspect({})` doesn't show the preview. Where as other calls like `util.isDate('a')` do show. Not sure `util.inspect` we can eager eval on.

Reply:
`util.inspect` can have side effects

Reply:
I intentionally implemented it like that: if the input is identical to the expected output, there's not much point in visualizing that as preview. These cases are mostly trivial and the user likely knows the outcome for input that does not change. I added a exception to that rule though. https://github.com/nodejs/node/pull/30811/files#diff-e9f927b0cccff85e4c622a17f6dfbd1fR194

Reply:
@targos Ahh, I got it. Thanks!